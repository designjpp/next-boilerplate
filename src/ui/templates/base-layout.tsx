import Link from "next/link";

import type { FC } from "react";

export const GBaseLayout: FC = ({ children }) => (
  <>
    <nav>
      <Link href="/">
        <a>Go to home</a>
      </Link>
      <Link href="/redux-sample">
        <a>Go to /redux-sample</a>
      </Link>
      <Link href="/redux-sample/server-side-props">
        <a>Go to /redux-sample/server-side-props</a>
      </Link>
    </nav>
    {children}

    <style jsx global>{`
      *:not(i) {
        font-family: Inter sans-serif;
      }

      #__next {
        @apply relative flex flex-col min-h-screen m-0 overflow-hidden text-base max-w-screen;

        padding: env(safe-area-inset-top) env(safe-area-inset-right)
          env(safe-area-inset-bottom) env(safe-area-inset-left);
      }
    `}</style>

    <style jsx>{`
      nav {
        @apply p-3 text-white bg-black;

        a {
          @apply p-3 mr-6;
        }
      }
    `}</style>
  </>
);
