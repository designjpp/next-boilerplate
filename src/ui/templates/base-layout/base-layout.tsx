import type { FC, CSSProperties } from 'react';

interface BaseLayoutProps {
	style?: CSSProperties;
	className?: string;
}

export const BaseLayout: FC<BaseLayoutProps> = ({
	style,
	className,
	children,
}) => (
	<>
		<main style={style} className={className}>
			{children}
		</main>
	</>
);
