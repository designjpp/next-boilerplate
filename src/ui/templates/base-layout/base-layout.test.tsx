import { create } from 'react-test-renderer';

import { BaseLayout } from './base-layout';

describe('<BaseLayout>', () => {
	it("should match is's snapshot", () => {
		const component = create(<BaseLayout />).toJSON();
		expect(component).toMatchSnapshot();
	});
});
