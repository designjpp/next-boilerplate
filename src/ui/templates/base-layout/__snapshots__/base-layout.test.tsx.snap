// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BaseLayout> should match is's snapshot 1`] = `<main />`;
